client/my-sites/plans-features-main/hooks/use-is-custom-domain-allowed-on-free-plan.ts

```diff
@@ -1,22 +1,21 @@
-import { getTld } from 'calypso/lib/domains';
 import { useExperiment } from 'calypso/lib/explat';
 import type { DataResponse } from 'calypso/my-sites/plan-features-2023-grid/types';
 
 const useIsCustomDomainAllowedOnFreePlan = (
 	flowName?: string | null,
-	domainName?: string
+	hasPaidDomainName?: boolean
 ): DataResponse< boolean > => {
-	const [ isLoadingAssignment, experimentAssignment ] = useExperiment(
-		'calypso_onboarding_plans_dotblog_on_free_plan_202307',
-		{
-			isEligible:
-				flowName === 'onboarding' && domainName != null && getTld( domainName ) === 'blog',
-		}
-	);
+	const EXPERIMENT_NAME =
+		flowName === 'onboarding'
+			? 'calypso_onboarding_plans_paid_domain_on_free_plan'
+			: 'calypso_onboardingpm_plans_paid_domain_on_free_plan';
```

Comment:
What flows go through here? Only `onboarding` and `onboarding-pm`? If other flows go in here, then everything not `onboarding` will get assigned to the `oboarding-pm` flow. Consider using `switch` to assign the experiment name.

Reply:
Actually we have an eligibility check on line 13 which prevents the experiment from being activated. This is why I thought adding a more explicit flow based check above would be redundant. I can however change it if you think it reads better ?

Reply:
Thanks for helping me understand! We can keep it as is, since the assignment filters for eligible users correctly.